src/ctap/clock.rs

```diff
@@ -0,0 +1,48 @@
+pub use embedded_time::Clock;
+use libtock_drivers::result::FlexUnwrap;
+use libtock_drivers::timer::Timer;
+
+pub struct LibtockClock<'a, const CLOCK_FREQUENCY: u32>(Timer<'a>);
+impl<'a, const CLOCK_FREQUENCY: u32> LibtockClock<'a, CLOCK_FREQUENCY> {
+    pub fn new(timer: Timer<'a>) -> Self {
+        Self(timer)
+    }
+}
+
+#[cfg(target_pointer_width = "32")]
+pub type ClockInt = u32;
+#[cfg(target_pointer_width = "64")]
+pub type ClockInt = u64;
+
+impl<const CLOCK_FREQUENCY: u32> embedded_time::Clock for LibtockClock<'_, CLOCK_FREQUENCY> {
+    type T = ClockInt;
+
+    const SCALING_FACTOR: embedded_time::fraction::Fraction =
+        <embedded_time::fraction::Fraction>::new(1, CLOCK_FREQUENCY);
+
+    fn try_now(&self) -> Result<embedded_time::Instant<Self>, embedded_time::clock::Error> {
+        let timer = &self.0;
+        let now = timer.get_current_clock().flex_unwrap();
+        Ok(embedded_time::Instant::new(now.num_ticks() as Self::T))
+    }
+}
+
+pub type CtapClock<'a> = LibtockClock<'a, 32768>;
+
+#[cfg(test)]
+pub const TEST_CLOCK_FREQUENCY_HZ: u32 = 32768;
+
+#[cfg(test)]
+#[derive(Default, Clone, Copy)]
+pub struct TestClock;
+
+#[cfg(test)]
```

Comment:
I would use `cfg(feature = "std")` instead, because fuzzing is not considered "test" by Rust. So we use the "std" feature to mean "the code is built for testing or fuzzing and is not production code".